packages/modal/src/resources/modal-util.ts

```diff
@@ -0,0 +1,28 @@
+import { UxModal } from '../ux-modal';
+
+export function findModal(item: HTMLElement | null): UxModal | null {
+  let element = item;
+  if (element === null) return null;
+  while (element.tagName !== 'BODY' && element.tagName !== 'UX-DRAWER') {
+    element = element.parentElement;
+    const isModaleElement = element ? modalViewModelFromElement(element) : false;
+    if (isModaleElement !== false) {
+      return isModaleElement;
+    }
+    if (element === null) return null;
+  }
+  return null;
+}
+
+export function modalViewModelFromElement(element: HTMLElement): UxModal | false {
```

Comment:
an `object | false` return type is quite a weird API to have. I think we can return `null` instead, and in the other check, it can be done against `null`, should be easier to guess, consistent with anywhere else, either in/out of Aurelia